spring-web/src/main/java/org/springframework/web/util/UriComponents.java

```diff
@@ -273,7 +271,15 @@ private static String getVariableName(String match) {
 	}
 
 	private static String getVariableValueAsString(@Nullable Object variableValue) {
-		return (variableValue != null ? variableValue.toString() : "");
+		if (variableValue != null) {
```

Comment:
might we have this as `if(variableValue == null) {  return ""; }`? and afterward keep evaluating the variable... so, I had in mind something like:
` if(Collection.class.isAssignableFrom(variableValue.getClass())){
      return ((Collection) variableValue).stream().map(Object::toString).collect(Collectors.joining(","));
} else {
   return variableValue.toString();
}`

so, we are safe from a NullPointerException and avoiding nested blocks, what makes the code cleaner, and using a little bit more streams collections features. 

https://docs.oracle.com/javase/8/docs/api/java/util/stream/Collectors.html#joining-java.lang.CharSequence

What do you think?